include/boost/fixed_string/detail/fixed_string.hpp

```diff
@@ -28,6 +29,96 @@ using is_input_iterator =
     std::integral_constant<bool,
         ! std::is_integral<T>::value>;
 
+template<std::size_t N>
+struct priority : priority<N - 1> { };
+
+template<>
+struct priority<0> { };
+
+template<std::size_t N, std::enable_if_t<N < std::numeric_limits<unsigned char>::max()>* = nullptr>
+constexpr unsigned char smallest_width_impl(priority<3>);
+
+template<std::size_t N, std::enable_if_t<N < std::numeric_limits<unsigned short>::max()>* = nullptr>
```

Comment:
calls to `max` have to be protected against macros, e.g.
```
(std::numeric_limits<unsigned short>::max)()
```

Or you could just write `USHRT_MAX` and spare yourself all of the lines included by `<numeric>`.